includes/class-wc-ajax.php

```diff
@@ -1229,8 +1229,26 @@ public static function json_search_customers() {
 			wp_die();
 		}
 
-		$data_store      = WC_Data_Store::load( 'customer' );
-		$ids             = $data_store->search_customers( $term );
+		// Stop if it is not numeric and smaller than 3 characters.
+		if ( ! is_numeric( $term ) && 2 >= strlen( $term ) ) {
```

Comment:
Still checking for less than 3 characters it's a burden generating long queries.